src/app/tests/index.test.ts

```diff
@@ -219,6 +227,50 @@ describe('src/index.ts', () => {
             });
     });
 
+    it('oauth/opensrp/callback works correctly if response is not stringfied JSON', async (done) => {
+        MockDate.set('1/1/2020');
+        JSON.parse = (body) => {
+            if (body === '{}') {
+                return 'string';
+            }
+        };
+        nock('http://reveal-stage.smartregister.org').get(`/opensrp/user-details`).reply(200, {});
+
+        request(app)
+            .get(oauthCallbackUri)
+            .end((err, res: request.Response) => {
+                panic(err, done);
+                expect(res.header.location).toEqual('/logout?serverLogout=true');
+                expect(res.notFound).toBeFalsy();
+                expect(res.redirect).toBeTruthy();
+                done();
+            });
+    });
+
+    it('logs user out from opensrp and calls keycloak', (done) => {
+        (fetch as any).mockImplementation(()=> Promise.resolve('successfull'));
+        request(app)
+            .get('/logout?serverLogout=true')
+            .set('Cookie', sessionString)
+            .end((err, res: request.Response) => {
+                panic(err, done);
+                expect(res.header.location).toEqual(`${EXPRESS_KEYCLOAK_LOGOUT_URL}?redirect_uri=${EXPRESS_SERVER_LOGOUT_URL}`);
+                expect(res.redirect).toBeTruthy();
+                expect(fetch).toHaveBeenCalledTimes(1);
+                expect(fetch).toHaveBeenCalledWith(
+                    EXPRESS_OPENSRP_LOGOUT_URL,
+                    {
+                        "headers": {
+                            "accept": "application/json",
+                            "authorization": "Bearer 64dc9918-fa1c-435d-9a97-ddb4aa1a8316",
```

Comment:
Where does the value of `64dc9918-fa1c-435d-9a97-ddb4aa1a8316` come from?

Reply:
It is the access token in fixtures.